lib/platform/linux.c

```diff
@@ -362,8 +362,11 @@ static int read_resp(struct switchtec_linux *ldev, void *resp,
 	}
 
 	memcpy(&ret, buf, sizeof(ret));
-	if (ret)
-		errno = ret;
+
+	if (ret) {
+		errno = ENODATA;
+		return -errno;
```

Comment:
I thought it was a little odd to be returning -errno here. I see it's already in this code above as well. Probably not that big a deal.